### PR TITLE
added package install instructions

### PR DESCRIPTION
**Pull request by ericlake**
initial version

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Sample Usage:
+#   This method should not be called directly.
+class oidentd::params {
+  case $::operatingsystem {
+    'RedHat', 'CentOS': {
+      case $::lsbmajdistrelease {
+        '6': {
+          $package_name = 'oidentd'
+        }
+      }
```
Comment (gregswift) [inline manifests/params.pp]: default: { fail('Only EL6 is supported in the RH family') }

File: manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#
+# === Authors
+#
+# Cloud Integration Operations <cit-ops@rackspace.com>
+#
+# === Copyright
+#
+# Copyright 2015 Rackspace
+#
+class oidentd {
```
Comment (gregswift) [inline manifests/init.pp]:
```
class (
  ensure = present,
  source  = undef,
  content = undef,
) {

  file { '/etc/oidentd.conf':
    ensure => $ensure,
    user =>,
    group =>,
    mode =>,
    source => $source,
    content => $content,
    notify     => Service['oidentd'],
}
  }

File: manifests/init.pp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# Copyright 2015 Rackspace
+#
+class oidentd (
+  ensure         = present,
+  service_ensure = running,
+  enable         = true,
+  source         = undef,
+  content        = undef,
+) {
+  include oidentd::params, oidentd::package, oidentd::service, oidentd::config
```
Comment (gregswift) [inline manifests/init.pp]: yeah, no.

separate lines pls.  Its also better to pass the parameters into the classes that need them, such as source and content into oidentd::config

File: manifests/config.pp [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class oidentd::config {
+  file { '/etc/oidentd.conf':
+    ensure  => $ensure,
+    owner   => 'root',
+    group   => 'root',
+    mode    => 0644,
+    source  => $source,
+    content => $content,
+    notify  => Class['oidentd::service'],
```
Comment (gregswift) [inline manifests/config.pp]: ``` notify => Service['oidentd'] ```

Notifying the class wont restart the service

File: manifests/service.pp [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class oidentd::service {
+  service { 'oidentd':
+    ensure => $service_ensure,
+    enable => $enable,
+    require => Class['oidentd::config']
```
Comment (gregswift) [inline manifests/service.pp]: ``` require => File[$oidentd::params::config_file] ```

File: manifests/config.pp [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class oidentd::config {
+  file { '/etc/oidentd.conf':
```
Comment (gregswift) [inline manifests/config.pp]: i'd add the file to the params

File: manifests/package.pp [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Requires:
+#   This module has no requirements.
+#
+# Sample Usage:
+#   This method should not be called directly.
+class oidentd::package {
+  package { $::oidentd::params::package_name:
+    ensure => present,
+    before => Class['oidentd::service']
```
Comment (gregswift) [inline manifests/package.pp]: this before is probably fine, but i personally do prefer to be more specific with what its before. although just saying, doesn't need to be updated necessarily

File: manifests/init.pp [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+#
+# Copyright 2015 Rackspace
+#
+class oidentd (
+  ensure         = present,
+  service_ensure = running,
+  enable         = true,
+) {
+  include oidentd::params,
+    oidentd::package,
```
Comment (gregswift) [inline manifests/init.pp]: close, but syntactically i think this is more appropriate

```
class oidentd (
  ensure               = present,
  service_ensure  = running,
  enable               = true,
  source               = undef,
  content              = undef,
) {
  include oidentd::params
  include oidentd::package
  class { 'oidentd::config':
    ensure => $ensure,
    source => $source,
    content => $content,
  }
  class { 'oidentd::service':
    ensure => $service_ensure;
    enable => $enable,
  }
}
```

Comment (ericlake) [inline manifests/init.pp]: Would I still have source and content as params in config.pp in this case?

File: manifests/config.pp [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+# Actions:
+#   This class installs oidentd package.
+#
+# Requires:
+#   This module has no requirements.
+#
+# Sample Usage:
+#   This method should not be called directly.
+class oidentd::config (
+  source         = undef,
```
Comment (gregswift) [inline manifests/config.pp]: the spacing for the ='s should be as far to the left as possible while still lining up with a minimium of 1 space between the parameter names and the =

File: manifests/service.pp [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Actions:
+#   This class installs oidentd package.
+#
+# Requires:
+#   This module has no requirements.
+#
+# Sample Usage:
+#   This method should not be called directly.
+class oidentd::service {
```
Comment (gregswift) [inline manifests/service.pp]: you dont have params here

File: manifests/config.pp [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+#
+# Actions:
+#   This class installs oidentd package.
+#
+# Requires:
+#   This module has no requirements.
+#
+# Sample Usage:
+#   This method should not be called directly.
+class oidentd::config inherits oidentd::params {
```
Comment (gregswift) [inline manifests/config.pp]:
```
class oidentd::config (
  ensure = present,
  source  = undef,
  content = undef,
) inherits oidentd::params {

  file { '/etc/oidentd.conf':
    ensure => $ensure,
    user =>,
    group =>,
    mode =>,
    source => $source,
    content => $content,
    notify     => Service['oidentd'],
  }
}
```

File: manifests/package.pp [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+# Actions:
+#   This class installs oidentd package.
+#
+# Requires:
+#   This module has no requirements.
+#
+# Sample Usage:
+#   This method should not be called directly.
+class oidentd::package (
+  ensure = $oidentd::params::ensure,
```
Comment (gregswift) [inline manifests/package.pp]: line up the ='s

File: manifests/config.pp [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+# Actions:
+#   This class installs oidentd package.
+#
+# Requires:
+#   This module has no requirements.
+#
+# Sample Usage:
+#   This method should not be called directly.
+class oidentd::config (
+  ensure = $oidentd::params::ensure,
```
Comment (gregswift) [inline manifests/config.pp]: line up the ='s

**Pull request merged**